index.bs

```diff
@@ -1353,21 +1335,22 @@ to {{SensorErrorEventInit}}.
         1.  Set |sensor_instance|.{{[[lastEventFiredAt]]}} to null.
 </div>
 
-<h3 dfn export>Revoke sensor permission</h3>
+<h3 dfn export>Generic Sensor permission revocation algorithm</h3>
 
-<div algorithm="revoke sensor permission">
+<div algorithm="generic sensor permission revocation algorithm">
 
     : input
-    :: |sensor|, a [=platform sensor=].
+    :: |permissionName|, a [=powerful feature/name|powerful feature name=]
     : output
     :: None
 
-    1.  Let |activated_sensors| be |sensor|'s associated [=ordered set|set=] of [=activated sensor objects=].
```

Comment:
I think this algorithm can still use **activated sensor objects** to good effect, but it needs to be able to translate _permissionName_ into a set of platform sensors by looking at each platform sensor's **sensor type** and checking if _permissionName_ is contained in its **sensor permission names**. This is what I think the original text meant by "associated".

Without this logic using it we can probably remove the **activated sensor objects** concept, though it is somewhat useful for defining how sensor readings are reused between `Sensor` instances.

Reply:
The problem's that it's not entirely clear what context the "permission revocation algorithm" runs in according to https://w3c.github.io/permissions/#reacting-to-revocation so I'm not sure what platform sensors should be iterated to start with. Should one start from the current realm and get a Window global and get the associated platform sensors? In this case, getting actual platform object instances that exist in a realm felt easier.

One thing I've done is add a check below to do nothing if a `sensor` is idle, so that we only act on active sensors (or sensors that are being activated).